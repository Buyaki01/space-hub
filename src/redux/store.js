import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import dragonsReducer from './dragons/dragonReducer';
<<<<<<< HEAD
import missionsReducer from './missions/missionsReducer';

const reducers = combineReducers({
  dragonsReducer,
  missionsReducer,
=======
import rocketsReducer from './rockets/rocketReducer';

const reducers = combineReducers({
  dragonsReducer,
  rocketsReducer,
>>>>>>> f8c89723c1210ea06b81e9f644d4ef10824c1ffd
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
