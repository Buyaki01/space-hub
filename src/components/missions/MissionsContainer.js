import MissionsList from './MissionsList';
// import './missions.css';

const MissionsContainer = () => (
  <div>
    <MissionsList />
  </div>
);
export default MissionsContainer;
