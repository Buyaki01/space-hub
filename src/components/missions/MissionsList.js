import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import MissionItem from './MissionItem';
import { getMissionsFromApi } from '../../redux/missions/missionsReducer';

const MissionsList = () => {
  const missionsArray = useSelector((state) => state.missionsReducer);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissionsFromApi());
  }, [dispatch]);
  return (
    <section>
      <table className="missionItemContainer">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
        </thead>

        {missionsArray.map((mission) => (
          <MissionItem key={mission.id} mission={mission} />
        ))}
      </table>
    </section>
  );
};
export default MissionsList;
