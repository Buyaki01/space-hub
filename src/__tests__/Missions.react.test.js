import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import MissionsContainer from '../components/missions/MissionsContainer';
import MissionsList from '../components/missions/MissionsList';
import store from '../redux/store';

describe('Missions', () => {
  test('should render component correctly', () => {
    const { asFragment } = render(<Provider store={store}><MissionsContainer /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render component correctly', () => {
    const { asFragment } = render(<Provider store={store}><MissionsList /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
