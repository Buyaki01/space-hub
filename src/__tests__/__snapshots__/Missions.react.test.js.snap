// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions should render component correctly 1`] = `
<DocumentFragment>
  <div>
    <section>
      <table
        class="missionItemContainer"
      >
        <thead>
          <tr
            class="border"
          >
            <th
              class="border"
            >
              Mission
            </th>
            <th
              class="border"
            >
              Description
            </th>
            <th
              class="border"
            >
              Status
            </th>
          </tr>
        </thead>
      </table>
    </section>
  </div>
</DocumentFragment>
`;

exports[`Missions should render component correctly 2`] = `
<DocumentFragment>
  <section>
    <table
      class="missionItemContainer"
    >
      <thead>
        <tr
          class="border"
        >
          <th
            class="border"
          >
            Mission
          </th>
          <th
            class="border"
          >
            Description
          </th>
          <th
            class="border"
          >
            Status
          </th>
        </tr>
      </thead>
    </table>
  </section>
</DocumentFragment>
`;
